### PR TITLE
Change type of RemoteCertificate property in TlsHandler to X509Certificate2

### PR DESCRIPTION
**Pull request by rajeevmv**

### CODE REVIEW

File: src/DotNetty.Handlers/Tls/TlsHandler.cs [hunk 1]
```diff
@@ -64,19 +64,13 @@ public TlsHandler(Func<Stream, SslStream> sslStreamFactory, TlsSettings settings
 
         public X509Certificate LocalCertificate => this.sslStream.LocalCertificate;
```
Comment (nayato) [inline src/DotNetty.Handlers/Tls/TlsHandler.cs]: can we do the same here?

File: src/DotNetty.Handlers/Tls/TlsHandler.cs [hunk 2]
```diff
@@ -64,19 +64,13 @@ public TlsHandler(Func<Stream, SslStream> sslStreamFactory, TlsSettings settings
...
 
-        public X509Certificate RemoteCertificate => this.sslStream.RemoteCertificate;
+        // using workaround mentioned here: https://github.com/dotnet/corefx/issues/4510
+        public X509Certificate2 RemoteCertificate => new X509Certificate2(this.sslStream.RemoteCertificate?.Export(X509ContentType.Cert));
```
Comment (nayato) [inline src/DotNetty.Handlers/Tls/TlsHandler.cs]: can you pls try safe casting to cert2 first and if that fails fallback to export?

File: src/DotNetty.Handlers/Tls/TlsHandler.cs (hunk 1 above)
Comment (rajeevmv) [inline src/DotNetty.Handlers/Tls/TlsHandler.cs]: done

File: src/DotNetty.Handlers/Tls/TlsHandler.cs (hunk 2 above)
Comment (rajeevmv) [inline src/DotNetty.Handlers/Tls/TlsHandler.cs]: done

**Pull request merged**